.github/workflows/CI.yml

```diff
@@ -147,150 +321,89 @@ jobs:
             upload-app: true
 
 
+  # -----------------------------------------------------------------------------------------------
   docker:
-    needs: [changelog-check, spell-check, upload-coverage]
+  # -----------------------------------------------------------------------------------------------
+    needs: [build-package]
     uses: ./.github/workflows/gcovr-ci-job.yml
     with:
       container: true
       gcc: ${{ matrix.gcc }}
-    secrets: inherit
+      python-version: 3.12
+      tag: ${{ needs.build-package.outputs.tag }}
+      tag_message: ${{ needs.build-package.outputs.tag_message }}
 
     strategy:
       fail-fast: false
       matrix:
         gcc: [gcc-5, gcc-6, gcc-8, gcc-9, gcc-10, gcc-11, gcc-12, gcc-13, gcc-14, clang-10, clang-13, clang-14, clang-15, clang-16]
 
-  validate-reports:
-    needs: [changelog-check, spell-check, upload-coverage]
-
+  # -----------------------------------------------------------------------------------------------
+  publish:
+  # -----------------------------------------------------------------------------------------------
+    needs: [build-package, documentation, host, docker]
```

Comment:
@Spacetown consider integrating https://github.com/marketplace/actions/alls-green#why as a CI gate that is possible to add to a branch protection ruleset as a single check with no need to manually enumerate all the jobs in the test matrix.